### ISSUE TITLE
[`large_types_passed_by_value`] does not trigger when the value is mutable

### ISSUE THREAD

**Issue by J-ZhengLi** [rust-lang/rust-clippy#11745]
### Summary

As title says~

### Lint Name

large_types_passed_by_value

### Reproducer

I tried this code:

```rust
#![allow(unused)]
#![deny(clippy::large_types_passed_by_value)]

#[derive(Clone, Copy)]
// Same as the example in its doc
struct TooLarge([u8; 2048]);

fn foo(mut v: TooLarge) {}

fn main() {}
```

I expected to see this happen:

warn me that the param `mut v` is too large,
suggesting me to pass `v: &mut TooLarge` instead

Instead, this happened:

no warning

### Version

```text
newest clippy version in playground:
0.1.75 (2023-11-01 75b064d)
```

**Comment by Jarcho** [comment]
This is done on purpose. Taking the argument by mutable reference would cause the modifications done by the function to be visible to the callee, and taking it by ref would require a clone.

**Comment by J-ZhengLi** [comment]
> This is done on purpose. Taking the argument by mutable reference would cause the modifications done by the function to be visible to the callee, and taking it by ref would require a clone.

I see now, thanks, I forgot that those two are different things.

~~Reminds me of a lot code I wrote using bevy engine requires passing `mut var_name` to functions... phew, almost digging a grave for myelf 😆~~
